service/history/shardContext.go

```diff
@@ -58,8 +58,8 @@ type (
 		GetConfig() *Config
 		GetLogger() bark.Logger
 		GetMetricsClient() metrics.Client
-		GetTimerAckLevel() time.Time
-		UpdateTimerAckLevel(ackLevel time.Time) error
+		GetTimerAckLevel(cluster string) time.Time
```

Comment:
Now we are going to have too many components managing ack levels separately.  Currently each update is a write to cassandra, this is not going to scale once cross dc support is in production.  Let's put a TODO to accept updates from individual components and have a separate job which updates all the ack levels in a single write for the entire shard.

Reply:
track in: https://github.com/uber/cadence/issues/634